src/hal/src/lib.rs

```diff
@@ -282,11 +282,30 @@ bitflags! {
         // Bits for Extensions
 
         /// Supports task shader stage.
-        const TASK_SHADER = 0x0000_0001 << 96;
+        const TASK_SHADER = 0x0001 << 96;
         /// Supports mesh shader stage.
-        const MESH_SHADER = 0x0000_0002 << 96;
+        const MESH_SHADER = 0x0002 << 96;
         /// Mask for all the features associated with mesh shader stages.
         const MESH_SHADER_MASK = Features::TASK_SHADER.bits | Features::MESH_SHADER.bits;
+
+        // Bits for downlevel hardware and apis
```

Comment:
We are sort of tight on flags right now, and want to start migrating things into better places. If I understand correctly, these are all flags, and they don't need to be "enabled" like regular Vulkan features. In this case, let's create a `struct DownlevelProperties`, and make it a member of `struct PhysicalDeviceProperties`. You can see there is already a few things in there :)

We'll probably want to make the `shader_model: u8` or an enum there instead of the flag, too.